core/Decode.cpp

```diff
@@ -76,24 +130,66 @@ namespace olympia
         uint32_t num_decode = std::min(uop_queue_credits_, fetch_queue_.size());
         num_decode = std::min(num_decode, num_to_decode_);
 
+        //buffer to maximize the chances of a group match limited
+        //by max allowed latency, bounded by max group size
+        if(fusion_enable_) {
+          if(num_decode < fusion_max_group_size_
+             && latency_count < fusion_max_latency_)
+          {
+            ++latency_count;
+            return;
+          }
+        }
+
+        latency_count = 0;
+
         if(num_decode > 0)
         {
             InstGroupPtr insts =
                 sparta::allocate_sparta_shared_pointer<InstGroup>(instgroup_allocator);
+
+            InstUidListType uids;
             // Send instructions on their way to rename
             for(uint32_t i = 0; i < num_decode; ++i) {
                 const auto & inst = fetch_queue_.read(0);
                 insts->emplace_back(inst);
                 inst->setStatus(Inst::Status::DECODED);
 
+                if(fusion_enable_) uids.push_back(inst->getMavisUid());
```

Comment:
Nit: please add `{}` around the `if` statement body.

Reply:
clang format did not correct this. I've fixed it manually.